vulcanai2/tests/models/conftest.py

```diff
@@ -0,0 +1,205 @@
+                    kernel_size=(5, 5),
+                    stride=2,
+                    pool_size=2,
+                    dropout=0.1
+                ),
+                dict(
+                    in_channels=24,
+                    out_channels=64,
+                    kernel_size=(5, 5),
+                    pool_size=2,
+                    dropout=0.1
+                )
+            ],
+        },
+        device='cpu'
+    )
+
+@pytest.fixture(scope="module")
+def conv3D_net():
+    """conv3D fixture."""
+    return ConvNet(
+        name='conv3D_net',
+        in_dim=(1, 28, 28, 28),
+        config={
+            'conv_units': [
+                dict(
+                    in_channels=1,
+                    out_channels=16,
+                    kernel_size=(5, 5, 5),
+                    stride=2,
+                    dropout=0.1
+                ),
+                dict(
+                    in_channels=16,
+                    out_channels=64,
+                    kernel_size=(5, 5, 5),
+                    dropout=0.1
+                )
+            ],
+        }
+    )
+
+@pytest.fixture(scope="module")
+def multi_input_dnn(conv1D_net, conv2D_net):
+    """Dense network fixture with two inputs."""
+    return DenseNet(
+        name='multi_input_dnn',
+        input_networks=[conv1D_net, conv2D_net],
+        config={
+            'dense_units': [100, 50],
+            'initializer': None,
+            'bias_init': None,
+            'norm': None,
+            'dropout': 0.5,  # Single value or List
+        },
+        device='cpu'
+    )
+
+@pytest.fixture(scope="module")
+def multi_input_cnn(conv3D_net, multi_input_dnn):
+    """Bottom multi-input network fixture."""
+    return ConvNet(
+        name='multi_input_cnn',
+        input_networks=[conv3D_net, multi_input_dnn],
+        num_classes=10,
+        config={
+            'conv_units': [
+                dict(
+                    in_channels=1,
+                    out_channels=16,
+                    kernel_size=(3, 3, 3),
+                    stride=2,
+                    dropout=0.1
+                ),
+            ],
+        },
+        device='cpu'
+    )
+@pytest.fixture(scope="module")
+def multi_input_dnn_data(conv1D_net, conv2D_net,
+                         multi_input_dnn):
+    return MultiDataset([
+        (
+            TensorDataset(
+                torch.ones([10, *conv1D_net.in_dim]),
+                torch.tensor([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]).long()),
+            True, True),
+        (
+            TensorDataset(torch.ones(
+                [10, *multi_input_dnn.input_networks['conv2D_net'].in_dim])),
+            True, False)
+    ])
+
+@pytest.fixture(scope="module")
+def multi_input_cnn_data(conv3D_net, multi_input_dnn_data):
+    return MultiDataset([
+        (TensorDataset(torch.ones([10, *conv3D_net.in_dim])), True, False),
+        multi_input_dnn_data
+    ])
+
+@pytest.fixture(scope="module")
+def cnn_noclass():
```

Comment:
I don't know if we need a conv2D_net AND a cnn_noclass. perhaps we can just pick one? lowers the number of fixtures to create?

Reply:
👍